api/v1/checks.go

```diff
@@ -220,6 +229,30 @@ func (c JmeterCheck) GetType() string {
 	return "jmeter"
 }
 
+// PopulateConnection will attempt to populate the host and port from the connection name.
+func (c *JmeterCheck) PopulateConnection(ctx context.Context, db *gorm.DB, k8sClient kubernetes.Interface, namespace string) error {
+	if c.ConnectionName == "" {
+		return nil
+	}
+
+	connection, err := duty.HydratedConnectionByURL(ctx, db, k8sClient, namespace, c.ConnectionName)
```

Comment:
The method name should be `HydrateConnectionByURL`

also not sure if this method should be on the API itself  - @yashmehrotra wdyt ?

Reply:
> The method name should be HydrateConnectionByURL

This, to me, implies that you already have a connection and you want to hydrate it.

```go
func HydrateConnectionByURL(ctx context.Context, connection Connection) {
    ...
}
```